File: interactor/templates/app-java/src/main/java/interactor/_Interactor.java
package <%= appPackage %>.domain.interactors.<%= interactorPackageName %>;

import rx.Scheduler;
import <%= appPackage %>.domain.interactors.base.BaseInteractor;

import javax.inject.Inject;
import rx.Observable;

public class <%= interactorName %>Interactor extends BaseInteractor {

    @Inject
    public <%= interactorName %>Interactor(Scheduler executor) {
        super(executor);
    }

    public Observable<Object> invoke() {
        return Observable.just(null);
    }

}

File: interactor/templates/app-kotlin/src/main/java/interactor/_Interactor.kt
package <%= appPackage %>.domain.interactors.<%= interactorPackageName %>

import <%= appPackage %>.domain.interactors.base.BaseCoroutineInteractor
import <%= appPackage %>.ui.base.BaseViewCoroutineScope
import <%= appPackage %>.domain.repository.<%= interactorPackageName %>.<%= interactorName %>LocalRepository
import <%= appPackage %>.domain.repository.<%= interactorPackageName %>.<%= interactorName %>RemoteRepository
import kotlinx.coroutines.CoroutineDispatcher

import javax.inject.Inject

class <%= interactorName %>Interactor @Inject constructor(dispatcher: CoroutineDispatcher) : BaseCoroutineInteractor(dispatcher)  {

    @Inject
    lateinit var remoteRepository: <%= interactorName %>RemoteRepository

    @Inject
    lateinit var localRepository: <%= interactorName %>LocalRepository
}
